### ISSUE TITLE
org-wide: Read MSRV from manifest in CI job

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3335]
MSRV should use SSOT. Instead of hardcoding the toolchain in the MSRV job we should read it from the manifest, reduces maintenance burden. 

To close this issue please do it in all repositories listed under `Core Crates`: https://github.com/rust-bitcoin

ref: https://github.com/rust-bitcoin/hex-conservative/pull/102#discussion_r1749556791

**Comment by Kixunil** [comment]
I was thinking of creating a directory called `project-metadata` containing workspace-wide configuration:

* MSRV
* maximum stable version
* nightly version
* all other pins

We could then have a CI checking that all crates follow project-wide MSRV etc. Note that *pulling* the information from there is still preferred but that is sometimes impossible. In version 1.64 we'll be able to pull information from workspace directly so we'll change that then.

**Comment by tcharding** [comment]
`cargo-rbmt` handles this correctly IIRC. Right @nyonson?

Perhaps we should have an issue that says 'use cargo-rbmt' in all core org crates'?

**Comment by nyonson** [comment]
Yep, it does what Kix says at the end there, it pulls the MSRV from package manifests and checks that we are actually testing with that toolchain.

> Perhaps we should have an issue that says 'use cargo-rbmt' in all core org crates'?

I think this is doable now, ironing out the last teething issues in rust-psbt.

**Comment by tcharding** [comment]
## All the core repos
- bitcoin: Done
- bech32: https://github.com/rust-bitcoin/rust-bech32/pull/234
- hex-conservative
- secp

## listed in https://github.com/rust-bitcoin but probably not worth doing
- bitcoinconsensus
- ordered

**Comment by apoelstra** [comment]
I think we can close this now as "will be done as we migrate to rbmt".
